fl4health/clients/basic_client.py

```diff
@@ -140,6 +150,30 @@ def _handle_logging(
             f"Client {metric_prefix} Losses: {loss_string} \n" f"Client {metric_prefix} Metrics: {metric_string}",
         )
 
+    def _handle_reporting(
+        self,
+        loss_dict: Dict[str, float],
+        metric_dict: Dict[str, Scalar],
+        current_round: Optional[int] = None,
+    ) -> None:
+
+        # If no current_round is passed or current_round is None, set current_round to 0
+        # This situation only arises when we do local finetuning and call train_by_epochs or train_by_steps explicitly
+        current_round = current_round if current_round is not None else 0
+
+        # We enforce that only one of self.total_epochs and self.total_steps is defined
+        # So only one of the two following lines will update the reporting dict
+        reporting_dict: Dict[str, Any] = {"server_round": current_round}
+        reporting_dict = (
+            {**reporting_dict, "epoch": self.total_epochs} if self.total_epochs is not None else reporting_dict
+        )
+        reporting_dict = (
+            {**reporting_dict, "step": self.total_steps} if self.total_steps is not None else reporting_dict
+        )
+        reporting_dict.update(loss_dict)
+        reporting_dict.update(metric_dict)
+        self._maybe_report_metrics(reporting_dict)
```

Comment:
Because logging and reporting were coupled together, we were forming this dictionary regardless of whether the reporter was defined or not. I think we can bring the `_maybe_report_metrics(...)` up into this class. So this function would be 
```
    def _maybe_report_metrics(
        self,
        loss_dict: Dict[str, float],
        metric_dict: Dict[str, Scalar],
        current_round: Optional[int] = None,
    ) -> None:
```
and we'd only do all of this work to create the dictionary if the reporter is not None. What do you think?

Reply:
Updated in [028422b](https://github.com/VectorInstitute/FL4Health/pull/61/commits/028422b3cc4efdce5bf2e37afcbc3ef98bfd4e24)!